### DISCUSSION TITLE
Calling Sigmoid in Train vs. Eval

### DISCUSSION THREAD

**Discussion by JRMARiA** [ultralytics/yolov5#9403] [Q&A] [answered]
Help me understand what I'm missing. 
In the Detect.forward() method in models/yolo.py we have: 
`if not self.training: 
    ... 
    y = x[i].sigmoid() 
    ... 
`
However, during training, we only call the sigmoid function on predictions that correspond to bounding box regression.
Is there something that I'm missing here? Why call sigmoid on box and class confidence if they are already bound between 0 and 1?

**Comment by JRMARiA** [comment, marked as answer]
I think I figured it out. Because we use BCE with Logits as loss, the obj/class predictions get sigmoided there.

**Reply by glenn-jocher** [reply]
@JRMARiA yes that's correct.
